lib/translator/inquiry.cc

```diff
@@ -0,0 +1,292 @@
+        formatted_hex_string[i] = '_';
+      }
+      formatted_hex_string[result.page_length - 1] = '.';
+
+      // insert numbers
+      int pos = 0;
+      for (int i = 0; i < result.page_length - 1; i++) {
+        if (formatted_hex_string[i] != '_') {
+          formatted_hex_string[i] = hex_string[pos++];
+        }
+      }
+
+      memcpy(product_serial_number, formatted_hex_string, result.page_length);
+    } else {
+      // 6.1.3.1.2
+      result.page_length = 20;
+
+      // convert 64-bit number into hex string
+      char hex_string[17];
+      sprintf(hex_string, "%08lx", identify_namespace_data.eui64);
+
+      // insert _ and . in the correct positions
+      char formatted_hex_string[21];
+      for (int i = 4; i < result.page_length; i += 5) {
+        formatted_hex_string[i] = '_';
+      }
+      formatted_hex_string[result.page_length - 1] = '.';
+
+      // insert numbers
+      int pos = 0;
+      for (int i = 0; i < result.page_length - 1; i++) {
+        if (formatted_hex_string[i] != '_') {
+          formatted_hex_string[i] = hex_string[pos++];
+        }
+      }
+
+      memcpy(product_serial_number, formatted_hex_string, result.page_length);
+    }
+  } else {
+    if (nguid_nz) {
+      // 6.1.3.1.1
+      result.page_length = 40;
+
+      // convert 128-bit number into hex string, 64-bits at a time
+      char hex_string[33];
+      sprintf(hex_string, "%08lx", identify_namespace_data.nguid[0]);
+      sprintf(&hex_string[16], "%08lx", identify_namespace_data.nguid[1]);
+
+      // insert _ and . in the correct positions
+      char formatted_hex_string[41];
+      for (int i = 4; i < result.page_length; i += 5) {
+        formatted_hex_string[i] = '_';
+      }
+      formatted_hex_string[result.page_length - 1] = '.';
+
+      // insert numbers
+      int pos = 0;
+      for (int i = 0; i < result.page_length - 1; i++) {
+        if (formatted_hex_string[i] != '_') {
+          formatted_hex_string[i] = hex_string[pos++];
+        }
+      }
+
+      memcpy(product_serial_number, formatted_hex_string, result.page_length);
+    } else {
+      // 6.1.3.1.3
+      // valid for NVMe 1.0 devices only
+      // TODO?
+    }
+  }
+
+  return result;
+}
+
+scsi_defs::UnitSerialNumber BuildUnitSerialNumberVpd() {
+  nvme_defs::IdentifyNamespace identify_namespace_data;
+  return TranslateUnitSerialNumberVpdResponse(identify_namespace_data);
+}
+
+// TODO: write return value to a buffer
+// Main logic engine for the Inquiry command
+void translate(absl::Span<const uint8_t> raw_cmd) {
+  scsi_defs::InquiryCommand cmd;
+  StatusCode status = RawToScsiCommand(raw_cmd, cmd);
+  if (status != StatusCode::kSuccess) return;
+
+  if (cmd.evpd) {
+    switch (cmd.page_code) {
+      case scsi_defs::PageCode::kSupportedVpd: {
+        // Return Supported Vpd Pages data page to application client, refer
+        // to 6.1.2.
+        scsi_defs::SupportedVitalProductData result = BuildSupportedVpdPages();
```

Comment:
It seems like the results here are not being used? if this is a placeholder for something, add a comment and/or TODO about it.

Reply:
waiting on a clear consensus on what we actually do with the result. at the top of the `translate() // TODO: write return value to a buffer`